### PR TITLE
provider/openstack: add created time to load balancer definition

### PR DESCRIPTION
**Pull request by drmaas**

### CODE REVIEW

**Comment by drmaas** [comment]
@spinnaker/reviewers ^^^

**Comment by drmaas** [comment]
hang on, seem to have an issue ...

**Comment by drmaas** [comment]
@spinnaker/reviewers @derekolk @varikin functionally tested the create and update cases, we're good to merge if all else looks good.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/domain/LoadBalancerResolver.groovy [hunk 1]
```diff
@@ -19,19 +19,81 @@ package com.netflix.spinnaker.clouddriver.openstack.domain
...
+   * @param description
+   * @return the port value
+   */
+  long parseCreatedTime(final String description) {
+    String s = match(description, createdPattern)
+    s ? s.toLong() : -1
+  }
+
+  /**
+   *
```
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/domain/LoadBalancerResolver.groovy]: Missing java doc comment.  Either add or remove.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/domain/LoadBalancerResolver.groovy [hunk 2]
```diff
@@ -19,19 +19,81 @@ package com.netflix.spinnaker.clouddriver.openstack.domain
...
+  /**
+   * Parse the internal port from a load balancer description in the following format.
+   * <br><br>
+   * {@code
+   *  ...,internal_port=8100,...
+   * }
+   * @param description
+   * @return the port value
+   */
+  int parseInternalPort(final String description) {
```
Comment (lwander) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/domain/LoadBalancerResolver.groovy]: This seems like a good time to use `Integer` and return `null` on an invalid input

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/model/OpenstackLoadBalancer.groovy [hunk 3]
```diff
@@ -28,7 +28,7 @@ import org.openstack4j.model.network.ext.HealthMonitor
 import org.openstack4j.model.network.ext.LbPool
 
 @Canonical
-@JsonIgnoreProperties(['nameRegex','namePattern','descriptionRegex','descriptionPattern'])
+@JsonIgnoreProperties(['nameRegex','namePattern','portRegex','portPattern','createdRegex','createdPattern'])
```
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/model/OpenstackLoadBalancer.groovy]: I believe nameRegex and Pattern are no longer being used ... can you please remove or I can follow up in another PR.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/domain/LoadBalancerResolver.groovy [hunk 4]
```diff
@@ -19,19 +19,81 @@ package com.netflix.spinnaker.clouddriver.openstack.domain
...
+  /**
+   * Parse the created time from a load balancer description in the following format.
+   * <br><br>
+   * {@code
+   *  ...,created_time=12345678,...
+   * }
+   * @param description
+   * @return the port value
+   */
+  long parseCreatedTime(final String description) {
```
Comment (lwander) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/domain/LoadBalancerResolver.groovy]: Same here as above comment.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/domain/LoadBalancerResolver.groovy (hunk 2 above)
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/domain/LoadBalancerResolver.groovy]: 👍

**Comment by lwander** [comment]
LGTM, up to you how you want to handle returning invalid inputs.

**Comment by derekolk** [comment]
LGTM 👍

**Pull request merged**